### PR TITLE
Feature connectivity tests

### PR DESCRIPTION
**Pull request by Davknapp**
Added two tests that check the correctness of the relations between type, local_id, cube_id and type of the parent implement in t8_dpyramid_connectivity.c
During testing I also found a small bug and fixed it. 

**_All these boxes must be checked by the reviewers before merging the pull request:_**

- [x] The author added a BSD statement to `doc/` (or already has one)
- [x] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)

  If the Pull request indroduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

- [x] All tests pass (in various configurations, this should be executed automatically in a github action)
- [x] New source/header files are properly added to the Makefiles
- [x] The reviewer executed the new code features at least once and checked the results manually
- [x] The code is covered in an existing or new test case
- [x] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [x] The code is well documented
- [x] All function declarations, structs/classes and their members have a proper doxygen documentation
- [x] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)
- [ ] Testing of this template: If you feel something is missing from this list, contact the developers

### CODE REVIEW

File: test/t8_schemes/t8_test_pyra_connectivity.cxx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      parent_type = t8_dpyramid_cid_type_to_parenttype[cid][type];
+      SC_CHECK_ABORTF (parent_type == pyra_parent_type,
+                       "Look-up tables are different");
+    }
+  }
+}
+
+/**
+ * Given the type of a parent element, iterate over all local ids of the children
+ * and look-up their type and cube-id. Usings these compute the parent-type again
```
Comment (sandro-elsweijer) [inline test/t8_schemes/t8_test_pyra_connectivity.cxx:57]:
```suggestion
 * and look-up their type and cube-id. Using these to compute the parent-type again
```

**Review by sandro-elsweijer — CHANGES_REQUESTED** [review]
I changed some values in the pyramid connectivity and almost every change resulted in a failing test, I am impressed about how good this test works.
@holke requested, that we implement the new tests as google test. Should we implement this test also as google test?

**Comment by Davknapp** [comment]
> I changed some values in the pyramid connectivity and almost every change resulted in a failing test, I am impressed about how good this test works. @holke requested, that we implement the new tests as google test. Should we implement this test also as google test?

Thank you!
I will transform the test into a google-test

**Review by sandro-elsweijer — COMMENTED** [review]
I was able to change the `1` in 
https://github.com/holke/t8code/blob/615500e272efd5d3a534d0b8fdbf4930935a01a2/src/t8_schemes/t8_default/t8_default_pyramid/t8_dpyramid_connectivity.c#L32
into a `2` without any test failing. Do you know why this happens?

**Comment by Davknapp** [comment]
> I was able to change the `1` in
> 
> https://github.com/holke/t8code/blob/615500e272efd5d3a534d0b8fdbf4930935a01a2/src/t8_schemes/t8_default/t8_default_pyramid/t8_dpyramid_connectivity.c#L32
> 
> 
> into a `2` without any test failing. Do you know why this happens?

Yes, if you use 
```t8_debugf("cid: %i, type: %i \n", cid, type);```
when p_type is 3 and Iloc is 4, you see that the cid is 6 and the type is 1 (2, if you change it in the lookup).

line 77 computes the check_type and in:
https://github.com/holke/t8code/blob/615500e272efd5d3a534d0b8fdbf4930935a01a2/src/t8_schemes/t8_default/t8_default_pyramid/t8_dpyramid_connectivity.c#L73
there are multiple 3. That is why the test does not fail.

Idealy we should implement tests for all look-up-tables in the connectivity-files.

**Comment by Davknapp** [comment]
> > I was able to change the `1` in
> > https://github.com/holke/t8code/blob/615500e272efd5d3a534d0b8fdbf4930935a01a2/src/t8_schemes/t8_default/t8_default_pyramid/t8_dpyramid_connectivity.c#L32
> > 
> > into a `2` without any test failing. Do you know why this happens?
> 
> Yes, if you use `t8_debugf("cid: %i, type: %i \n", cid, type);` when p_type is 3 and Iloc is 4, you see that the cid is 6 and the type is 1 (2, if you change it in the lookup).
> 
> line 77 computes the check_type and in:
> 
> https://github.com/holke/t8code/blob/615500e272efd5d3a534d0b8fdbf4930935a01a2/src/t8_schemes/t8_default/t8_default_pyramid/t8_dpyramid_connectivity.c#L73
> 
> 
> there are multiple 3. That is why the test does not fail.
> Idealy we should implement tests for all look-up-tables in the connectivity-files.

The problem is, that the current test does not test the arrays independantly. Theoratically one could change all three arrays such that the test works fine, but the arrays themself aren't correct anymore. Do you have an idea to check the arrays independantly (without creating another look-up-table at the test)?

**Review by sandro-elsweijer — APPROVED** [review]

**Pull request merged**